src/Data/Array/ST.purs

```diff
@@ -146,12 +141,10 @@ thaw = copyImpl
 foreign import toAssocArray """
   function toAssocArray(arr) {
     return function(){
-      var as = [];
-      var i = -1;
-      var n = arr.length;
-      while(++i < n) {
+      var n = arr.length >>> 0;
+      var as = new Array(n);
+      for (var i = 0 >>> 0; i < n; i++)
```

Comment:
Looks like >>>0 is indeed the hint for unsigned.

Take a look at the first benchmark in http://mbebenita.github.io/LLJS/

Change the loop limit to something like 0x7fffffff in all 3 tests, it seems int is the fastest type.